### PR TITLE
Submits Ticket 9 GET/api/reviews/review_id/comments for sign off

### PR DESCRIPTION
**Pull request by stef-lukes**
# 9. GET/api/reviews/review_id/comments 

- Creates controller and model files thus separating concerns for comments
- Utilises Promises.all in model file although I had thought about using the fetchReviewById function in the controller as well, or creating a separate utility function.
- Successfully tests for two 200 paths, one with custom error handling
- Successfully test for 400 and 404 errors

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -273,3 +273,55 @@ describe("8 GET /api/reviews", () => {
...
+  });
+  test("status 400, invalid data type (not a number) passed into query as review_id", () => {
+    return request(app)
+      .get("/api/reviews/invalid_id/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid request");
+      });
+  });
+  test("status 200, valid path but review has no comments and returns empty array ", () => {
```
Comment (popatre) [inline __tests__/app.test.js:318]: Test says responds with empty array - actual tests doesnt test this

File: __tests__/app.test.js [hunk 2]
```diff
@@ -273,3 +273,55 @@ describe("8 GET /api/reviews", () => {
...
+    return request(app)
+      .get("/api/reviews/1/comments")
+      .expect(200)
+      .then((res) => {
+        const { body } = res;
+        expect(body.msg).toBe(`Review 1 has no comments`);
```
Comment (popatre) [inline __tests__/app.test.js:324]: Should be an empty array

File: app.js [hunk 3]
```diff
@@ -37,7 +40,7 @@ app.use((err, req, res, next) => {
 });
 
 app.use((err, req, res, next) => {
-  console.log(err);
+  // console.log(err);
```
Comment (popatre) [inline app.js:43]: remove log

**Review by popatre — CHANGES_REQUESTED** [review]
Just a change to the test/function needed

File: app.js [hunk 4]
```diff
@@ -40,7 +40,7 @@ app.use((err, req, res, next) => {
 });
 
 app.use((err, req, res, next) => {
-  // console.log(err);
+  console.log(err, "<< 500 error");
```
Comment (popatre) [inline app.js:43]: remove log

File: __tests__/app.test.js [hunk 5]
```diff
@@ -315,7 +315,7 @@ describe.only("9. GET /api/reviews/:review_id/comments", () => {
         expect(body.msg).toBe("Invalid request");
       });
   });
-  test("status 200, valid path but review has no comments and returns empty array ", () => {
+  test("status 200, valid path but review has no comments. Should return custom error handler.", () => {
```
Comment (popatre) [inline __tests__/app.test.js:318]: It should return an empty array. The name was correct, the function/test was wrong.

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js (hunk 5 above)
Comment (stef-lukes) [reply __tests__/app.test.js:318]: understood ✔️

File: __tests__/app.test.js [hunk 6]
```diff
@@ -273,3 +273,55 @@ describe("8 GET /api/reviews", () => {
...
+              body: expect.any(String),
+              votes: expect.any(Number),
+              author: expect.any(String),
+              review_id: 2,
+              created_at: expect.any(String),
+            })
+          );
+        });
+      });
+  });
```
Comment (bainesface) [inline __tests__/app.test.js:301]: Hello! Try to be consistent with your destructuring. In the test above you have 
```js
    .then((res) => {
        const { comments } = res.body;
```
but below 

```js
.then(({ body }) => {
```

either is fine, just keep it the same throughout

**Review by Rosem10 — APPROVED** [review]

**Pull request merged**